pkg/sources/adapter/azureservicebussource/adapter.go

```diff
@@ -230,22 +243,56 @@ func connectionStringFromEnvironment(namespace, entityPath string) string {
 //  Both (DataAction):
 //  - Microsoft.ServiceBus/namespaces/messages/receive/action
 func (a *adapter) Start(ctx context.Context) error {
+	maxMessages := 100
 	logging.FromContext(ctx).Info("Listening for messages")
 
-	handle := a.msgRcvr.Listen(ctx, servicebus.HandlerFunc(a.handleMessage))
-	<-handle.Done()
-	return handle.Err()
+loop:
+	for {
+		select {
+		case <-ctx.Done():
+			break loop
+		default:
+			messages, err := a.msgRcvr.ReceiveMessages(ctx, maxMessages, nil)
+			if err != nil {
+				return fmt.Errorf("error receiving messages: %w", err)
+			}
+			for _, m := range messages {
+				body, err := m.Body()
+				if err != nil {
+					return fmt.Errorf("error getting message data: %w", err)
+				}
+
+				msg := &Message{
+					Data:            body,
+					ReceivedMessage: *m,
+					LockToken:       stringifyLockToken((*uuid.UUID)(&m.LockToken)),
+				}
```

Comment:
(related to a previous conversation that is now marked a "Resolved").

Regardless of the signature of the `handleMessage()` function, it would be nice to keep the code organized.
The original intention was to have all the data processing inside `process.go`, otherwise the received message is mangled in different places, making the whole message processing logic hard to follow.

Keeping everything inside `process.go` is as simple as creating a function that performs the conversion from the native format to the custom `*Message`:

```go
func toMessage(rcvMsg *servicebus.ReceivedMessage) (*Message, err) {
	body, err := m.Body()
	if err != nil {
		return nil, fmt.Errorf("reading message data: %w", err)
	}

	return &Message{
		Data:            body,
		ReceivedMessage: rcvMsg,
		LockToken:       stringifyLockToken((*uuid.UUID)(&m.LockToken)),
	}, nil
}
```

Reply:
Should be done now, thanks!